### DISCUSSION TITLE
Error in readme demo script

### DISCUSSION THREAD

**Discussion by litotes18** [ha-pu/globaltrends#144] [Q&A]
Hi, just heard your fantastic talk at useR2021!
Trying to recreate your workflow on on your readme at github
https://github.com/ha-pu/globaltrends
I see the following error:

compute_score(control = new_control, object = new_object)
Error in .f(.x[[i]], ...) : 
  Error: Batch number must be object of type integer.
You provided a non-integer value.
I just changed the line
new_object <- add_object_keyword(keyword = c("wind turbine", "rooftop solar", "e-mobility"))
to three other objects...
Could you please tell me, why this error is occurring?
Thanks!

**Comment by ha-pu** [comment]
Thanks for notifying me! I found the issue and will fix that asap. In the meantime, you can replace `control = new_control` by `control = unlist(new_control)` or just write `control = 1`.

**Comment by ha-pu** [comment]
I've fixed the problem. If you reinstall `globaltrends`, the problem should not occur again.

**Comment by litotes18** [comment]
Hallo,

danke für die schnelle Antwort + Fix. Tatsächlich läuft das jetzt durch.
Allerdings stoppt das Skript schon ein bisschen später, nämlich bei

plot_bar(data = out_score)
Error in unique(data$keyword)[[1]] : subscript out of bounds

Ich nehme an, das liegt daran, dass alle out_ Objekte leer sind (0 observations of x vars).
Könntest du da noch mal nachgucken, bitte?

Danke!

Ulrich

Von: ha-pu ***@***.***>
Gesendet: Dienstag, 6. Juli 2021 20:25
An: ha-pu/globaltrends ***@***.***>
Cc: Frey, Ulrich ***@***.***>; Author ***@***.***>
Betreff: Re: [ha-pu/globaltrends] Error in readme demo script (#144)


I've fixed the problem. If you reinstall globaltrends, the problem should not occur again.

**Reply by ha-pu** [reply]
Also bei mir ist alles durchgelaufen, ich habe meinen Code angehängt.

Dumme Frage, kann es sein, dass du die Filter Optionen im `export_score` Call nicht angepasst hast?

[run_readme.txt](https://github.com/ha-pu/globaltrends/files/6775647/run_readme.txt)

**Comment by litotes18** [comment]
Hi,
also jetzt bin ich baff. Ich habe deinen Code 1:1 kopiert und nichts geändert…
Kriege aber immer noch denselben Fehler

plot_bar(data = out_score)
Error in unique(data$keyword)[[1]] : subscript out of bounds

Ich habe R 4.0.2 und eine neu RStudio-Version.

Ulrich

P.S:

Dumme Frage, kann es sein, dass du die Filter Optionen im export_score Call nicht angepasst hast?
Damit meinst du, die Keywords aus
new_object <- add_object_keyword(keyword = c("manchester united", "real madrid"))
in
out_score <- export_score(keyword = "manchester united")
zu übertragen, oder? Können das auch mehr als eines sein?

Von: ha-pu ***@***.***>
Gesendet: Mittwoch, 7. Juli 2021 10:34
An: ha-pu/globaltrends ***@***.***>
Cc: Frey, Ulrich ***@***.***>; Author ***@***.***>
Betreff: Re: [ha-pu/globaltrends] Error in readme demo script (#144)


Also bei mir ist alles durchgelaufen, ich habe meinen Code angehängt.

Dumme Frage, kann es sein, dass du die Filter Optionen im export_score Call nicht angepasst hast?

run_readme.txt<https://github.com/ha-pu/globaltrends/files/6775647/run_readme.txt>

**Reply by ha-pu** [reply]
Das ist tatsächlich sehr sonderbar. Die `export_score` und `export_voi` Funktionen funktionieren auch mit mehreren Keywords:
 ```
 out_score <- export_score(keyword = c("wind turbine", "rooftop solar", "e-mobility"))
out_voi <- export_voi(keyword = c("wind turbine", "rooftop solar", "e-mobility"))
```

Könntest du mal kontrollieren, ob in der Datenbank auch wirklich Daten stehen? Kommt bei dir mit diesem Befehl das selbe Ergebnis heraus?
```
> count(tbl_score, keyword)
# Source:   lazy query [?? x 2]
# Database: sqlite 3.35.5 [\\SHARES\data\Users\hpuhr\Docs\Documents\db\globaltrends_db.sqlite]
  keyword           n
  <chr>         <int>
1 e-mobility     8976
2 rooftop solar  8976
3 wind turbine   8976
```
Falls nicht, könntest du bitte die auch die *control* Downloads `count(tbl_control, keyword)` und die *object* Downloads `count(tbl_object, keyword)` kontrollieren?

**Comment by litotes18** [comment]
Hallo,

angehängt mal das Ergebnis deiner Fragen im Screenshot.
Control + Object sind voll, keyword count leer…

Ulrich

Von: ha-pu ***@***.***>
Gesendet: Mittwoch, 7. Juli 2021 13:14
An: ha-pu/globaltrends ***@***.***>
Cc: Frey, Ulrich ***@***.***>; Author ***@***.***>
Betreff: Re: [ha-pu/globaltrends] Error in readme demo script (#144)


Das ist tatsächlich sehr sonderbar. Die export_score und export_voi Funktionen funktionieren auch mit mehreren Keywords:

out_score <- export_score(keyword = c("wind turbine", "rooftop solar", "e-mobility"))



out_voi <- export_voi(keyword = c("wind turbine", "rooftop solar", "e-mobility"))



Könntest du mal kontrollieren, ob in der Datenbank auch wirklich Daten stehen? Kommt bei dir mit diesem Befehl das selbe Ergebnis heraus?

> count(tbl_score, keyword)



# Source:   lazy query [?? x 2]



# Database: sqlite 3.35.5 [\\SHARES\data\Users\hpuhr\Docs\Documents\db\globaltrends_db.sqlite]



  keyword           n



  <chr>         <int>



1 e-mobility     8976



2 rooftop solar  8976



3 wind turbine   8976



Falls nicht, könntest du bitte die auch die control Downloads count(tbl_control, keyword) und die object Downloads count(tbl_object, keyword) kontrollieren?

**Reply by ha-pu** [reply]
Ich sehe den Screenshot zwar leider nicht, aber wenn nur in `data_control` und `data_object` Daten stehen, kommen wir dem Problem eventuell auf die Spur. Probiere bitte die search scores nochmals zu berechnen `compute_score(control = 1, object = 1)`. Hoffentlich klappt es dann.

**Comment by litotes18** [comment]
Hi,
Screenshot war angehängt – hier noch einmal.
Ergebnis:
…
Successfully computed search score | control: 1 | object: 1 | location: LK [67/67]
out_score <- export_score(keyword = "manchester united") = OK (läuft durch)

Aber wieder:
plot_bar(data = out_score)
Error in unique(data$keyword)[[1]] : subscript out of bounds

Grund (vermutlich): out_score Objekt ist ein leerer Dataframe (0 obs of 8 variables), ebenso wie out_doi und out_voi.

Ulrich

Von: ha-pu ***@***.***>
Gesendet: Donnerstag, 8. Juli 2021 09:56
An: ha-pu/globaltrends ***@***.***>
Cc: Frey, Ulrich ***@***.***>; Author ***@***.***>
Betreff: Re: [ha-pu/globaltrends] Error in readme demo script (#144)


Ich sehe den Screenshot zwar leider nicht, aber wenn nur in data_control und data_object Daten stehen, kommen wir dem Problem eventuell auf die Spur. Probiere bitte die search scores nochmals zu berechnen compute_score(control = 1, object = 1). Hoffentlich klappt es dann.

**Reply by ha-pu** [reply]
Hier ist offenbar das Problem: `out_score <- export_score(keyword = "manchester united")`. Die Funktion exportiert nur Werte für das Keyword "manchester united". Aber dafür gibt es ja keine Daten. Was passiert, wenn du den keyword-Filter in diesen Befehl anpasst?
```
out_score <- export_score(keyword = c("wind turbine", "rooftop solar", "e-mobility"))
out_voi <- export_voi(keyword = c("wind turbine", "rooftop solar", "e-mobility"))
```

Ich sehe den Screenshot leider wieder immer nicht. Wenn es noch immer nicht klappt, kannst du ihn mir bitte direkt an harald.puhr@wu.ac.at schicken?
